kategory-core/src/test/kotlin/kategory/data/OptionTTest.kt

```diff
@@ -2,17 +2,26 @@ package kategory
 
 import io.kotlintest.KTestJUnitRunner
 import io.kotlintest.properties.forAll
-import kategory.laws.FunctorFilterLaws
-import kategory.laws.TraverseFilterLaws
 import io.kotlintest.matchers.shouldNotBe
 import org.junit.runner.RunWith
 
 @RunWith(KTestJUnitRunner::class)
 class OptionTTest : UnitSpec() {
+
     val EQ_ID: Eq<HK<OptionTKindPartial<IdHK>, Int>> = Eq { a, b ->
         a.value() == b.value()
     }
 
+    val EQ_OPTION = object : Eq<HK<OptionTKindPartial<OptionHK>, Int>> {
+        override fun eqv(a: HK<OptionTKindPartial<OptionHK>, Int>, b: HK<OptionTKindPartial<OptionHK>, Int>): Boolean =
+                a.ev().value == b.ev().value
+    }
+
+    val EQ_NESTED_OPTION = object : Eq<HK<OptionTKindPartial<OptionHK>, HK<OptionTKindPartial<OptionHK>, Int>>> {
```

Comment:
It may be that you could make the three Eq instances into a single method parametrized in T.

Reply:
I have added 

```kotlin
fun <T> EQ(): Eq<T> = Eq { a, b ->
        a.value() == b.value()
    }
```

but it doesn't work


```
kategory/kategory-core/src/test/kotlin/kategory/data/OptionTTest.kt: (12, 24): Unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public fun <A, T> ConstKind<???, ???> /* = HK<HK<ConstHK, ???>, ???> */.value(): ??? defined in kategory
public fun <A> IdKind<???> /* = HK<IdHK, ???> */.value(): ??? defined in kategory
public inline fun <F, A> OptionTKind<???, ???> /* = HK<HK<OptionTHK, ???>, ???> */.value(): HK<???, Option<???>> defined in kategory
public inline fun <F, W, A> WriterTKind<???, ???, ???> /* = HK<HK2<[ERROR : Unknown type parameter 3], [ERROR : Unknown type parameter 4], [ERROR : Unknown type parameter 5]> /* = HK<HK<WriterTHK, ???>, ???> */, ???> */.value(): HK<???, Tuple2<???, ???>> defined in kategory
```

I don't know why ヽ(。_°)ノ

Reply:
It has to be Eq<HK<OptionTKindPartial<T>>. Let's chat about why in Slack.